conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/Endpoint.java

```diff
@@ -49,6 +50,11 @@
     /** Simple name of the endpoint method. This data may be used for metric instrumentation. */
     String name();
 
+    /** Is present if the method is deprecated, and contains its corresponding deprecating documentation. */
+    default Optional<String> deprecated() {
```

Comment:
:+1: in Jersey-land, this information would be loaded using reflection, but since we're not doing any reflection it looks nice to add it here.